docs/rules-engine/how-to-use/custom-fact.md

```diff
@@ -0,0 +1,125 @@
+data specific to your application and your business. 
+For this reason, the Otter framework does not provide any facts but the [implicit built-in facts](./built-in-facts.md).
+
+Let's see how to create a custom fact.
+
+## Implementation
+
+Facts are managed in services that extend the `FactsService` provided by `@o3r/rules-engine`. The service
+exposes a `fact` map linking the key of each fact exposed in the service to their respective value, which can either
+be an instance, a `Promise` or an `Observable` of the fact.
+
+The facts exposed by the service are defined in a dedicated interface that extends `FactDefinitions`. 
+
+You can generate the fact structure thanks to the `ng g facts-service [name] -- --path=[path-to-my-fact-folder] ` command.
+
+This will result in two files:
+* The fact definition
+```typescript
+import type { FactDefinitions } from '@o3r/rules-engine';
+
+export interface CustomFacts extends FactDefinitions {
+  fact1: boolean;
+  fact2: string;
+}
+
+```
+* The service fact
+```typescript
+import {FactsService} from '@o3r/rules-engine';
+import type {CustomFacts} from './custom.facts.ts';
+
+@Injectable()
+export class CustomFactsService extends FactsService<CustomFacts> {
+
+  public facts = {};
+
+}
+```
+
+In order to use the fact, you need to define its logic. It can be based on manual update, on the selection of a subset
+of state, on services etc.
+
+For this example, the `fact1` will be based on a store and the `fact2` will be set once and will reflect the user's browser 
+agent.
+
+```typescript
+import {FactsService} from '@o3r/rules-engine';
+import {Store} from '@ngrx/store';
+import {distinctUntilChanged, filter, map, select} from 'rxjs/operators';
+import {selectUserSelection} from '../store/user-selection';
+import type {SelectionState} from '../store/user-selection';
+import type {CustomFacts} from './custom.facts.ts';
+
+@Injectable()
+export class CustomFactsService extends FactsService<CustomFacts> {
+
+  constructor(store: Store<SelectionState>) {
+    super();
+    this.facts = {
+      fact1: store.pipe(
+        filter((store) => !!store.userSelection),
+        select(selectUserSelection),
+        map((userSelection) => !!userSelection),
+        distinctUntilChanged()
+      ),
+      fact2: window.navigator.userAgent
+    }
+  }
+}
+
+```
+
+> [!INFO]
+> There are two interesting things to notice about `fact1`:
+> * There is a pipe with a filter operator to ensure that the state is defined, before using it. This permits to register 
+> the fact even if the store is lazy loaded later in the flow. This observable will not emit until the store is loaded, but
+> rules engine will create the first emission (undefined) while waiting for the fact to emit its first value.
+> * Facts based on observable should always contain a `distinctUntilChange()` to avoid the re-evaluation of all the rules
+that depend on them.
```

Comment:
formatting

Reply:
This is markdown, the back to line is just to avoid too long lines on the file. This won't appear in the documentation.